coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CalciteTrinoUDFOperatorTransformers.java

```diff
@@ -0,0 +1,220 @@
+    operatorTransformers.add(createOperatorTransformer(
+        daliToCalciteOp("com.linkedin.stdudfs.lookup.hive.BrowserLookup"), 3, "browserlookup"));
+    addDaliFunctionId(daliFunctionIdSet, "com.linkedin.stdudfs.lookup.hive.BrowserLookup", 3);
+    operatorTransformers.add(createOperatorTransformer(
+        daliToCalciteOp("com.linkedin.jobs.udf.hive.ConvertIndustryCode"), 1, "converttoindustryv1"));
+    addDaliFunctionId(daliFunctionIdSet, "com.linkedin.jobs.udf.hive.ConvertIndustryCode", 1);
+    operatorTransformers.add(createOperatorTransformer(
+        daliToCalciteOp("com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper"), 1, "urn_extractor"));
+    addDaliFunctionId(daliFunctionIdSet, "com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper", 1);
+    operatorTransformers.add(createOperatorTransformer(
+        daliToCalciteOp("com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper"), 1, "urn_extractor"));
+    addDaliFunctionId(daliFunctionIdSet, "com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper", 1);
+
+    addDaliUDFs(operatorTransformers, daliFunctionIdSet);
+
+    //ad-hoc transformers
+    operatorTransformers.add(new MapStructAccessOperatorTransformer());
+
+    return operatorTransformers;
+  }
+
+  private static SqlOperator hiveToCalciteOp(String functionName) {
+    Collection<Function> lookup = HIVE_REGISTRY.lookup(functionName);
+    // TODO: provide overloaded function resolution
+    return lookup.iterator().next().getSqlOperator();
+  }
+
+  private static void addDaliFunctionId(Set<String> daliFunctionIdSet, String daliFunctionName, int operandNum) {
+    daliFunctionIdSet.add(daliFunctionName + "_" + operandNum);
+  }
+
+  private static void addDaliUDFs(List<SqlCallTransformer> operatorTransformers, Set<String> daliFunctionIdSet) {
+    ImmutableMultimap<String, Function> registry = HIVE_REGISTRY.getRegistry();
+    Converter<String, String> caseConverter = CaseFormat.UPPER_CAMEL.converterTo(CaseFormat.LOWER_UNDERSCORE);
+    for (Map.Entry<String, Function> entry : registry.entries()) {
+      // we cannot use entry.getKey() as function name directly, because keys are all lowercase, which will
+      // fail to be converted to lowercase with underscore correctly
+      final String hiveFunctionName = entry.getValue().getFunctionName();
+      if (!hiveFunctionName.startsWith("com.linkedin")) {
+        continue;
+      }
+      String[] nameSplit = hiveFunctionName.split("\\.");
+      // filter above guarantees we've at least 2 entries
+      String className = nameSplit[nameSplit.length - 1];
+      String funcName = caseConverter.convert(className);
+      SqlOperator op = entry.getValue().getSqlOperator();
+      for (int i = op.getOperandCountRange().getMin(); i <= op.getOperandCountRange().getMax(); i++) {
+        if (!daliFunctionIdSet.contains(hiveFunctionName + "_" + i)) {
```

Comment:
Should be `funcName` instead of `hiveFunctionName` (caught by the result comparison in regression test).

Reply:
Done.